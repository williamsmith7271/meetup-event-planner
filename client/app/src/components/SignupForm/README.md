## SignupForm Component
A signup form component.

### Example

```js
<SignupForm
  {...fields}
  onSubmit={this.handleSubmit}
/>
```

### Props

| Prop          | Type     | Default     | Possible Values
| ------------- | -------- | ----------- | ---------------------------------------------
| **onSubmit**    | Func   |             | A callback function called on submission


### Other Information
See index.js for the props and example
