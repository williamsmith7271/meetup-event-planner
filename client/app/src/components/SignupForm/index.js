import React, { PropTypes } from 'react';
import styles from './index.module.scss';
import cssModules from 'react-css-modules';
import Form from 'grommet-udacity/components/Form';
import Heading from 'grommet-udacity/components/Heading';
import FormField from 'grommet-udacity/components/FormField';
import FormFields from 'grommet-udacity/components/FormFields';
import Footer from 'grommet-udacity/components/Footer';
import Button from 'grommet-udacity/components/Button';
import Box from 'grommet-udacity/components/Box';
import { AuthFormFooter } from 'components';
import calculatedError from 'utils/error';

const SignupForm = ({
  onSubmit,
  nameInput,
  emailInput,
  passwordInput,
  passwordConfirmationInput,
  bioInput,
  employerInput,
  invalid,
}) => (
  <Box
    size={{ width: { max: 'large' } }}
    className={styles.signupForm}
    pad={{ horizontal: 'large' }}
  >
    <Form>
      <Heading strong align="center">
        Meetup Event Planner
      </Heading>
      <Heading align="center" tag="h5">
        Signup
      </Heading>
      <FormFields>
        <FormField
          help="What should we call you?"
          error={calculatedError(nameInput)}
          label="Name *"
          htmlFor="nameInput"
          className={styles.formField}
        >
          <input
            {...nameInput}
            required
            autoFocus
            id="nameInput"
            name="name"
            type="text"
            className={styles.input}
          />
        </FormField>
        <FormField
          help="How should we get in touch with you?"
          error={calculatedError(emailInput)}
          label="Email *"
          htmlFor="emailInput"
          className={styles.formField}
        >
          <input
            {...emailInput}
            required
            id="emailInput"
            name="email"
            type="email"
            className={styles.input}
          />
        </FormField>
        <FormField
          help="Make it secure"
          error={calculatedError(passwordInput)}
          label="Password *"
          htmlFor="passwordInput"
          className={styles.formField}
        >
          <input
            {...passwordInput}
            required
            id="passwordInput"
            name="password"
            type="password"
            className={styles.input}
          />
        </FormField>
        <FormField
          help="Confirm your password"
          label="Password Confirmation *"
          htmlFor="passwordConfirmationInput"
          className={styles.formField}
          error={calculatedError(passwordConfirmationInput)}
        >
          <input
            {...passwordConfirmationInput}
            required
            id="passwordConfirmationInput"
            type="password"
            name="password"
            className={styles.input}
          />
        </FormField>
        <FormField
          label="Bio"
          className={styles.formField}
          help="Optional Bio for Profile"
          htmlFor="bio-input"
        >
          <textarea
            {...bioInput}
            id="bio-input"
            name="bio"
            type="text"
            rows="3"
            cols="40"
          />
        </FormField>
        <FormField
          label="Employer"
          className={styles.formField}
          help="Optional Employer Field"
          htmlFor="bio-input"
        >
          <input
            {...employerInput}
            name="organization"
            id="employer-input"
            type="text"
          />
        </FormField>
      </FormFields>
      <Footer pad={{ vertical: 'medium' }} align="center">
        <Button onClick={invalid ? null : onSubmit} fill label="Submit" primary />
      </Footer>
      <AuthFormFooter text="Already a member?" link="/login" />
    </Form>
  </Box>
);

SignupForm.propTypes = {
  onSubmit: PropTypes.func.isRequired,
  nameInput: PropTypes.object.isRequired,
  emailInput: PropTypes.object.isRequired,
  passwordInput: PropTypes.object.isRequired,
  passwordConfirmationInput: PropTypes.object.isRequired,
  bioInput: PropTypes.object.isRequired,
  employerInput: PropTypes.object.isRequired,
  invalid: PropTypes.bool.isRequired,
};

export default cssModules(SignupForm, styles);
