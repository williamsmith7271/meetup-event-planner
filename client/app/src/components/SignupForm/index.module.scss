.signupForm {
  display: flex;
  align-items: center;
  justify-content: center;
  background-color: #fff;
  padding: 24px;
}

.formField {
  max-width: 430px;
}
