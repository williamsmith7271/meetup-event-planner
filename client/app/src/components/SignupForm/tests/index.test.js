import SignupForm from '../index';
import { shallow } from 'enzyme';
import { shallowToJson } from 'enzyme-to-json';
import React from 'react';

describe('<SignupForm />', () => {
  it('should render with default props', () => {
    const fields = {
      nameInput: {
        touched: false,
        error: null,
      },
      emailInput: {
        touched: false,
        error: null,
      },
      passwordInput: {
        touched: false,
        error: null,
      },
    };
    const wrapper = shallow(
      <SignupForm {...fields} onSubmit={(e) => e} />
    );
    expect(shallowToJson(wrapper)).toMatchSnapshot();
  });
});
