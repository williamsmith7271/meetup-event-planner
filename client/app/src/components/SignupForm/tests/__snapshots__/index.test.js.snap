exports[`<SignupForm /> should render with default props 1`] = `
<Box
  announce={false}
  direction="column"
  focusable={true}
  pad={
    Object {
      "horizontal": "large"
    }
  }
  responsive={true}
  tag="div">
  <Form
    compact={false}
    fill={false}
    pad="none">
    <Heading
      align="center"
      strong={true}
      tag="h1">
      Meetup Event Planner
    </Heading>
    <Heading
      align="center"
      tag="h5">
      Signup
    </Heading>
    <undefined
      children={
        <function Box() {
            (0, _classCallCheck3.default)(this, Box);
            return (0, _possibleConstructorReturn3.default)(this, (Box.__proto__ || (0, _getPrototypeOf2.default)(Box)).apply(this, arguments));
          }
          announce={false}
          direction="column"
          focusable={true}
          pad="none"
          responsive={true}
          size="medium"
          tag="div">
          <function Heading() {
              (0, _classCallCheck3.default)(this, Heading);
              return (0, _possibleConstructorReturn3.default)(this, (Heading.__proto__ || (0, _getPrototypeOf2.default)(Heading)).apply(this, arguments));
            }
            align="center"
            tag="h4">
            Secure Password Tips
          </function Heading() {
              (0, _classCallCheck3.default)(this, Heading);
              return (0, _possibleConstructorReturn3.default)(this, (Heading.__proto__ || (0, _getPrototypeOf2.default)(Heading)).apply(this, arguments));
            }>
          <function Heading() {
              (0, _classCallCheck3.default)(this, Heading);
              return (0, _possibleConstructorReturn3.default)(this, (Heading.__proto__ || (0, _getPrototypeOf2.default)(Heading)).apply(this, arguments));
            }
            align="center"
            tag="h5">
            Password must have at least
          </function Heading() {
              (0, _classCallCheck3.default)(this, Heading);
              return (0, _possibleConstructorReturn3.default)(this, (Heading.__proto__ || (0, _getPrototypeOf2.default)(Heading)).apply(this, arguments));
            }>
          <ul>
            <li>
              <function Paragraph() {
                  (0, _classCallCheck3.default)(this, Paragraph);
                  return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
                }
                className={undefined}>
                1. Eight characters total
              </function Paragraph() {
                  (0, _classCallCheck3.default)(this, Paragraph);
                  return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
                }>
            </li>
            <li>
              <function Paragraph() {
                  (0, _classCallCheck3.default)(this, Paragraph);
                  return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
                }
                className={undefined}>
                2. One uppercase character
              </function Paragraph() {
                  (0, _classCallCheck3.default)(this, Paragraph);
                  return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
                }>
            </li>
            <li>
              <function Paragraph() {
                  (0, _classCallCheck3.default)(this, Paragraph);
                  return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
                }
                className={undefined}>
                3. One special character
              </function Paragraph() {
                  (0, _classCallCheck3.default)(this, Paragraph);
                  return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
                }>
            </li>
            <li>
              <function Paragraph() {
                  (0, _classCallCheck3.default)(this, Paragraph);
                  return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
                }
                className={undefined}>
                4. One numerical character
              </function Paragraph() {
                  (0, _classCallCheck3.default)(this, Paragraph);
                  return (0, _possibleConstructorReturn3.default)(this, (Paragraph.__proto__ || (0, _getPrototypeOf2.default)(Paragraph)).apply(this, arguments));
                }>
            </li>
          </ul>
        </function Box() {
            (0, _classCallCheck3.default)(this, Box);
            return (0, _possibleConstructorReturn3.default)(this, (Box.__proto__ || (0, _getPrototypeOf2.default)(Box)).apply(this, arguments));
          }>
      }
      isShowing={true}
      onClose={[Function onPasswordBlur]} />
    <FormFields>
      <FormField
        error="value Required"
        help="What should we call you?"
        htmlFor="nameInput"
        label="Name *"
        size="medium"
        strong={false}>
        <input
          active={false}
          autoFocus={true}
          autofilled={false}
          dirty={false}
          error="value Required"
          id="nameInput"
          initialValue=""
          invalid={true}
          name="name"
          pristine={true}
          required={true}
          touched={true}
          type="text"
          valid={false}
          value=""
          visited={true} />
      </FormField>
      <FormField
        error={null}
        help="How should we get in touch with you?"
        htmlFor="emailInput"
        label="Email *"
        size="medium"
        strong={false}>
        <input
          active={false}
          autofilled={false}
          dirty={false}
          error="value Required"
          id="emailInput"
          initialValue=""
          invalid={true}
          name="email"
          pristine={true}
          required={true}
          touched={false}
          type="email"
          valid={false}
          value=""
          visited={false} />
      </FormField>
      <FormField
        error={null}
        help="Start typing for password tips"
        htmlFor="passwordInput"
        label="Password *"
        size="medium"
        strong={false}>
        <input
          active={false}
          autofilled={false}
          dirty={false}
          error="value Required"
          id="passwordInput"
          initialValue=""
          invalid={true}
          name="password"
          onBlur={[Function onPasswordBlur]}
          onClick={[Function onPasswordFocus]}
          onFocus={[Function onPasswordFocus]}
          pristine={true}
          required={true}
          touched={false}
          type="password"
          valid={false}
          value=""
          visited={false} />
      </FormField>
      <FormField
        error={null}
        help="Confirm your password"
        htmlFor="passwordConfirmationInput"
        label="Password Confirmation *"
        size="medium"
        strong={false}>
        <input
          active={false}
          autofilled={false}
          dirty={false}
          error="value Required"
          id="passwordConfirmationInput"
          initialValue=""
          invalid={true}
          name="password"
          pristine={true}
          required={true}
          touched={false}
          type="password"
          valid={false}
          value=""
          visited={false} />
      </FormField>
      <FormField
        help="Optional Bio for Profile"
        htmlFor="bio-input"
        label="Bio"
        size="medium"
        strong={false}>
        <textarea
          active={false}
          autofilled={false}
          cols="40"
          dirty={false}
          id="bio-input"
          initialValue=""
          invalid={false}
          name="bio"
          pristine={true}
          rows="3"
          touched={false}
          type="text"
          valid={true}
          value=""
          visited={false} />
      </FormField>
      <FormField
        help="Optional Employer Field"
        htmlFor="bio-input"
        label="Employer"
        size="medium"
        strong={false}>
        <input
          active={false}
          autofilled={false}
          dirty={false}
          id="employer-input"
          initialValue=""
          invalid={false}
          name="organization"
          pristine={true}
          touched={false}
          type="text"
          valid={true}
          value=""
          visited={false} />
      </FormField>
    </FormFields>
    <Footer
      align="center"
      direction="row"
      pad={
        Object {
          "vertical": "medium"
        }
      }
      responsive={false}>
      <Button
        fill={true}
        label="Submit"
        onClick={null}
        primary={true}
        type="button" />
    </Footer>
    <undefined
      link="/login"
      text="Already a member?" />
  </Form>
</Box>
`;
